File: angular-frontend/src/app/models/agreement.ts
export class Agreement {

  id?:number;
  workFunction:string;
  grossSalary:number;
  employmentHours:number;
  agreementHours:number;

  constructor( workFunction: string, grossSalary: number, employmentHours: number, agreementHours: number,id?: number) {
    this.id = id;
    this.workFunction = workFunction;
    this.grossSalary = grossSalary;
    this.employmentHours = employmentHours;
    this.agreementHours = agreementHours;
  }
}

File: springboot-backend/src/main/java/com/example/springbootbackend/domain/Agreement.java
package com.example.springbootbackend.domain;


import jakarta.persistence.*;
import lombok.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Agreement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //werkfunctie  (bediende, arbeider,...
    private String workFunction;

    //brutoloon
    private int grossSalary;

    //aantal uren tewerkstelling (hoeveel uren dat employee effectief werkt  bv 30h
    private int employmentHours;

    //totaal contract uren (bv 38 h werkweek)
    private int agreementHours;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "employee_id")
    private Employee employee;

}
